demos/quiz_bot/screens.py

```diff
@@ -0,0 +1,249 @@
+"""The module contains the screens the bot consists of."""
+
+import json
+import random
+
+import aiofiles
+
+from hammett.conf import settings
+from hammett.core import Button, Screen
+from hammett.core.constants import SourceTypes
+from hammett.core.handlers import register_button_handler
+from hammett.core.mixins import StartMixin
+from hammett.utils.translation import gettext as _
+from hammett.widgets import MultiChoiceWidget, SingleChoiceWidget
+from hammett.widgets.base import BaseChoiceWidget
+
+
+class BaseScreen(Screen):
+    """The base screen for all the screens in the bot."""
+
+    hide_keyboard = True
+
+    caption = ''
+
+    async def get_description(self, _update, context):
+        """Return the translated description of the screen."""
+        language_code = context.user_data.get('language_code', 'en')
+        return _(self.caption, language_code)
+
+    @staticmethod
+    def get_next_choice_widget(next_correct_answer):
+        """Return the widget screen according to the type of answer to
+        the next question.
+        """
+        return (
+            QuizMultiChoiceWidget
+            if isinstance(next_correct_answer, list)
+            else QuizSingleChoiceWidget
+        )
+
+
+class BaseQuizScreen(BaseScreen, BaseChoiceWidget):
+    """The base screen for all the screens participating in the quiz."""
+
+    async def add_extra_keyboard(self, _update, context):
+        """Add an extra keyboard below the widget buttons."""
+        language_code = context.user_data.get('language_code', 'en')
+        return [[
+            Button(
+                _('🏠 Main Menu', language_code),
+                MainMenuScreen,
+                source_type=SourceTypes.MOVE_SOURCE_TYPE),
+            Button(
+                _('Next ➡️', language_code),
+                self.next_question,
+                source_type=SourceTypes.HANDLER_SOURCE_TYPE),
+        ]]
+
+    async def get_choices(self, _update, context, **_kwargs):
+        """Return the choices with the available answers to the question."""
+        index = context.user_data['question_index']
+        questions = context.user_data['questions']
+
+        result = [(answer, answer) for answer in questions[index]['answers']]
+        random.shuffle(result)
+
+        return tuple(result)
+
+    async def get_description(self, _update, context):
+        """Return the `description` attribute of the screen."""
+        index = context.user_data['question_index']
+        questions = context.user_data['questions']
+        language_code = context.user_data.get('language_code', 'en')
+
+        return (
+            f'№ <b>{index + 1}/{len(context.user_data["questions"])}.</b> '
+            f'{questions[index]["question"][language_code]}'
+        )
+
+    @register_button_handler
+    async def next_question(self, update, context):
+        """Check the answer and switch to the next question or to the screen
+        with the results of the completed quiz.
```

Comment:
`Check the answer and switch to the next question or to the results screen for the completed quiz.`